pinot-core/src/main/java/org/apache/pinot/core/operator/transform/transformer/datetimehop/EpochToEpochWindowHopTransformer.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.core.operator.transform.transformer.datetimehop;
+
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.apache.pinot.spi.data.DateTimeFormatSpec;
+import org.apache.pinot.spi.data.DateTimeGranularitySpec;
+
+public class EpochToEpochWindowHopTransformer extends BaseDateTimeWindowHopTransformer<long[], long[][]> {
+  public EpochToEpochWindowHopTransformer(@Nonnull DateTimeFormatSpec inputFormat,
+                                          @Nonnull DateTimeFormatSpec outputFormat,
+                                          @Nonnull DateTimeGranularitySpec outputGranularity,
+                                          @Nonnull DateTimeGranularitySpec hopSize) {
+    super(inputFormat, outputFormat, outputGranularity, hopSize);
+  }
+
+  @Override
+  public void transform(@Nonnull long[] input, @Nonnull long[][] output, int length) {
+    for (int i = 0; i < length; i++) {
+      long epochTime = input[i];
+      long millisSinceEpoch = transformEpochToMillis(epochTime);
+      List<Long> hopWindows = hopWindows(millisSinceEpoch);
+      output[i] = hopWindows
```

Comment:
Suggest not using stream API for performance sensitive path

Reply:
Refactored it by using for loop